Add Google App Engine Compatibility

Google App Engine has wrapped Go in their own goapp binary, which includes references for appengine packages. When I try and run ffjson on my project, it throws errors when those are referenced.

```
STDERR:
..\..\mjibson\appstats\appstats.go:29:2: cannot find package "appengine" in any of:
        C:\Go\src\pkg\appengine (from $GOROOT)
        C:\google-cloud-sdk\platform\google_appengine\gopath\src\appengine (from $GOPATH)
..\..\mjibson\appstats\appstats.go:30:2: cannot find package "appengine/memcache" in any of:
        C:\Go\src\pkg\appengine\memcache (from $GOROOT)
        C:\google-cloud-sdk\platform\google_appengine\gopath\src\appengine\memcache (from $GOPATH)
..\..\mjibson\appstats\appstats.go:31:2: cannot find package "appengine/user" in any of:
        C:\Go\src\pkg\appengine\user (from $GOROOT)
        C:\google-cloud-sdk\platform\google_appengine\gopath\src\appengine\user (from $GOPATH)
..\..\mjibson\appstats\appstats.go:32:2: cannot find package "appengine_internal" in any of:
        C:\Go\src\pkg\appengine_internal (from $GOROOT)
        C:\google-cloud-sdk\platform\google_appengine\gopath\src\appengine_internal (from $GOPATH)
```

--- a/ffjson.go
+++ b/ffjson.go
@@ -29,6 +29,7 @@ import (
 )
 
 var outputPathFlag = flag.String("w", "", "Write generate code to this path instead of ${input}_ffjson.go.")
+var goCmdFlag = flag.String("go-cmd", "", "Path to go command; Useful for `goapp` support.")
 
 func usage() {
 	fmt.Fprintf(os.Stderr, "Usage of %s:\n\n", os.Args[0])
@@ -51,13 +52,20 @@ func main() {
 	inputPath := filepath.ToSlash(extra[0])
 
 	var outputPath string
-	if outputPathFlag == nil {
-		outputPath = *outputPathFlag
-	} else {
+	if outputPathFlag == nil || *outputPathFlag == "" {
 		outputPath = extRe.ReplaceAllString(inputPath, "${1}_ffjson.go")
+	} else {
+		outputPath = *outputPathFlag
 	}
 
-	err := generator.GenerateFiles(inputPath, outputPath)
+	var goCmd string
+	if goCmdFlag == nil || *goCmdFlag == "" {
+		goCmd = "go"
+	} else {
+		goCmd = *goCmdFlag
+	}
+
+	err := generator.GenerateFiles(goCmd, inputPath, outputPath)
 
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Error: %s:\n\n", err)

--- a/generator/generator.go
+++ b/generator/generator.go
@@ -22,13 +22,13 @@ import (
 	"fmt"
 )
 
-func GenerateFiles(inputPath string, outputPath string) error {
+func GenerateFiles(goCmd string, inputPath string, outputPath string) error {
 	packageName, structs, err := ExtractStructs(inputPath)
 	if err != nil {
 		return err
 	}
 
-	im := NewInceptionMain(inputPath, outputPath)
+	im := NewInceptionMain(goCmd, inputPath, outputPath)
 
 	err = im.Generate(packageName, structs)
 	if err != nil {

--- a/generator/inceptionmain.go
+++ b/generator/inceptionmain.go
@@ -61,14 +61,16 @@ type templateCtx struct {
 }
 
 type InceptionMain struct {
+	goCmd        string
 	inputPath    string
 	outputPath   string
 	TempMainPath string
 	tempMain     *os.File
 }
 
-func NewInceptionMain(inputPath string, outputPath string) *InceptionMain {
+func NewInceptionMain(goCmd string, inputPath string, outputPath string) *InceptionMain {
 	return &InceptionMain{
+		goCmd:      goCmd,
 		inputPath:  inputPath,
 		outputPath: outputPath,
 	}
@@ -165,7 +167,7 @@ func (im *InceptionMain) Run() error {
 	var out bytes.Buffer
 	var errOut bytes.Buffer
 
-	cmd := exec.Command("go", "run", "-a", im.TempMainPath)
+	cmd := exec.Command(im.goCmd, "run", "-a", im.TempMainPath)
 	cmd.Stdout = &out
 	cmd.Stderr = &errOut
 

